### PR TITLE
fix: deprecated code in new credential sync endpoints

### LINKED ISSUE #14115: API builds failing on main
@emrysal @joeauyeung seems like #13963 got merged after #14069 which makes some of this code obsolete and causing failures. Pushing a fix soon.

_Originally posted by @zomars in https://github.com/calcom/cal.com/issues/13963#issuecomment-2002055283_

### PR DESCRIPTION
**Pull request by zomars**
## What does this PR do?

Fixes #14115

## Type of change

- Bug fix (non-breaking change which fixes an issue)

## How should this be tested?

- `yarn build --filter=@calcom/api`

## Mandatory Tasks

- [x] Make sure you have self-reviewed the code. A decent size PR without self-review might be rejected.

### CODE REVIEW

File: apps/api/pages/api/credential-sync/_delete.ts [hunk 1]
```diff
@@ -40,12 +41,7 @@ import { schemaCredentialDeleteParams } from "~/lib/validations/credential-sync"
  *        description: Credential syncing not enabled
  */
 async function handler(req: NextApiRequest) {
-  const { prisma } = req;
-
-  const { userId: reqUserId, credentialId: reqCredentialId } = schemaCredentialDeleteParams.parse(req.query);
-
-  const userId = parseInt(reqUserId);
-  const credentialId = parseInt(reqCredentialId);
```
Comment (zomars) [inline apps/api/pages/api/credential-sync/_delete.ts:48]: The schema already returns as number.

File: apps/api/pages/api/credential-sync/_post.ts [hunk 2]
```diff
@@ -88,26 +88,29 @@ async function handler(req: NextApiRequest) {
...
-    include: {
-      user: {
-        select: {
-          email: true,
-        },
-      },
+    select: credentialForCalendarServiceSelect,
+  });
+  // createdcredential.user.email;
+  // TODO:              ^ Investigate why this select doesn't work.
```
Comment (zomars) [inline apps/api/pages/api/credential-sync/_post.ts:101]: @joeauyeung  Found some odd behavior in here. Seems like using a select or include in here doesn't actually returns the correct types. Tried replicating in other files and on a new repo with no luck. Also tried upgrading prisma to discard a possible bug but still didn't work. I had to create and find in two queries as a workaround.

**Review by zomars — COMMENTED** [review]
Self review done

**Review by alishaz-polymath — APPROVED** [review]
LGTM

**Pull request merged**